workflows/hcp_pnl_topup.lsf

```diff
@@ -1,84 +1,75 @@
 #!/usr/bin/bash
 
-# Useful wiki for executing this in PNL GPU machines
-# https://github.com/pnlbwh/luigi-pnlpipe/wiki/Run-HCP-pipeline-on-PNL-GPU-machines-in-a-parallel-manner
+# Copy this script to your project directory and edit as needed
+# There are no arguments to this script. Execute it as ./hcp_pnl_topup.lsf
 
-: << COMMENT
-Running this script w/o GPU support would take a long time (>24 hours).
-So always try to use a GPU enabled machine e.g. grx** nodes, eristwo gpu queue.
-=> Copy this script to your project directory and update the marked variables.
-=> Execute within shell as ./hcp_pnl_topup.lsf
-=> To execute through LSF, adjust [1-N]%2 and "BSUB -n 4" as explained below.
-COMMENT
+# Useful wiki for executing this script in PNL GPU machines
+# https://github.com/pnlbwh/luigi-pnlpipe/wiki/Run-HCP-pipeline-on-PNL-GPU-machines
 
-# ==============================================================================
-
-#BSUB -J hcp-topup[1-N]%2
-#BSUB -q gpu
-#BSUB -m ml001
-#BSUB -R rusage[mem=12000]
-#BSUB -o /data/pnl/U01_HCP_Psychosis/data_processing/output/hcp-topup-%J-%I.out
-#BSUB -e /data/pnl/U01_HCP_Psychosis/data_processing/output/hcp-topup-%J-%I.err
-#BSUB -n 4
+: << COMMENT
+HCP toolbox accepts an even number (2 or 4) of opposing templates.
+You can write four templates in the following order:
 
-# check LSB_JOBINDEX because it won't exist in /rfanfs/
-if [ ! -z ${LSB_JOBINDEX} ]
-then
-    export CUDA_VISIBLE_DEVICES=$(( ${LSB_JOBINDEX}%2 ))
-fi
+    PA template PA template
+    AP template AP template
 
-: << COMMENT
-The formula for CUDA_VISIBLE_DEVICES is ${LSB_JOBINDEX}%G,
-which also means the maximum number of parallel cases you can process is G
+Or two templates in the following order:
 
-Adjust "BSUB -n 4" in a way that each GPU device can run no more than one job.
-You can use the formula "BSUB -n N/G" to ensure that where--
-    - N is the maximum number of jobs for that node
-    - G is the number of GPUs in that node
-Otherwise your jobs might crash due to out of memory error.
+    PA template AP template
 
-Example: node ml001 has 8(=N) job slots and 2(=G) GPUs so "BSUB -n 8/2" and "BSUB -J hcp-topup[1-N]%2"
+Notes:
+    * Even if one of the templates is a single b0, name it like DWI with _dwi.nii.gz suffix.
+    * All b0 volumes need to be accompanied by .bval and .bvec files. Usually, the .bvec file is in 3xN format.
```

Comment:
The above comments clarify how single pair of DWI, or one DWI+one b0 could be provided to this script.